File: imports-frontend-entities/src/entities/billing_information.js
const handler = require('./base/handler')
const PostalAddress = require('./postal_address')

module.exports = class BillingInformation {

  constructor(obj = {}) {

    this.isConfirmed = obj.isConfirmed
    this.emailAddress = obj.emailAddress
    this.phoneNumber = obj.phoneNumber
    this.contactName = obj.contactName
    this.postalAddress = obj.postalAddress ? new PostalAddress(obj.postalAddress) : undefined

    return Object.seal(new Proxy(this, handler))
  }
}

File: notification-schema-java/src/main/java/uk/gov/defra/tracesx/notificationschema/representation/BillingInformation.java
package uk.gov.defra.tracesx.notificationschema.representation;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import javax.validation.Valid;
import javax.validation.constraints.NotBlank;
import lombok.Builder;
import lombok.Data;
import lombok.extern.jackson.Jacksonized;

@Builder
@Data
@JsonInclude(Include.NON_EMPTY)
@Jacksonized
public class BillingInformation {

  @NotBlank
  private Boolean isConfirmed;

  @NotBlank
  private String emailAddress;

  @NotBlank
  private String phoneNumber;

  @NotBlank
  private String contactName;

  @Valid
  private PostalAddress postalAddress;
}
